jquery.overlay.js

```diff
@@ -211,23 +216,40 @@
           // Style attribute's string
           style = 'background-color:' + strategy.css['background-color'];
 
-          text.contents().each(function () {
-            var text, html, str, prevIndex;
-            if (this.nodeType != Node.TEXT_NODE) return;
-            text = this.textContent;
+          if (this.allowOverlapping) {
+            var textContent, prevIndex, str, html;
+            textContent = text.html()
             html = '';
             for (prevIndex = match.lastIndex = 0;; prevIndex = match.lastIndex) {
-              str = match.exec(text);
+              str = match.exec(textContent);
               if (!str) {
-                if (prevIndex) html += escape(text.substr(prevIndex));
+                if (prevIndex) html += textContent.substr(prevIndex);
                 break;
               }
               str = str[0];
-              html += escape(text.substr(prevIndex, match.lastIndex - prevIndex - str.length));
-              html += '<span style="' + style + '">' + escape(str) + '</span>';
+              html += textContent.substr(prevIndex, match.lastIndex - prevIndex - str.length);
+              html += '<span style="' + style + '">' + str + '</span>';
             };
-            if (prevIndex) $(this).replaceWith(html);
-          });
+            if (prevIndex) text.html(html);
+          } else {
+            text.contents().each(function () {
+              var text, html, str, prevIndex;
+              if (this.nodeType != Node.TEXT_NODE) return;
+              text = this.textContent;
+              html = '';
+              for (prevIndex = match.lastIndex = 0; ; prevIndex = match.lastIndex) {
```

Comment:
okay, so i'm pretty sure we won't hit an infinite loop (in my version, i think this is true of the original too but for a different reason) because for each strategy, we're not updating the string that we're checking for matches until we're done looking for matches. that said, i should point out that there will be other problems with this if you start trying to match stuff in the html, but it'll be pretty obvious that it doesn't work, i don't think we have plans to do that, and i'm not actually sure there's a way to have this functionality without accepting that or doing a bunch more manipulation (which would explain why it wasn't originally built with it.)

i think it's probably best to go ahead with this (post review and QA) and file improvements to be done if they come up or on my own time, unless either of you thinks it's particularly important to have now. :)